client/src/main/java/io/pravega/client/stream/impl/StreamSegments.java

```diff
@@ -71,13 +75,20 @@ public StreamSegments withReplacementRange(StreamSegmentsWithPredecessors replac
         verifyReplacementRange(replacementRanges);
         NavigableMap<Double, Segment> result = new TreeMap<>();
         Map<Long, List<SegmentWithRange>> replacedRanges = replacementRanges.getReplacementRanges();
-        for (Entry<Double, Segment> exitingSegment : segments.entrySet()) {
-            List<SegmentWithRange> replacements = replacedRanges.get(exitingSegment.getValue().getSegmentId());
+        for (Entry<Double, Segment> existingSegment : segments.entrySet()) {
+            List<SegmentWithRange> replacements = replacedRanges.get(existingSegment.getValue().getSegmentId());
             if (replacements == null || replacements.isEmpty()) {
-                result.put(exitingSegment.getKey(), exitingSegment.getValue());
+                result.put(existingSegment.getKey(), existingSegment.getValue());
             } else {
                 for (SegmentWithRange replacement : replacements) {
-                    result.put(replacement.getHigh(), replacement.getSegment());
+                    /*
```

Comment:
We also need to add a check and test for the case where (using the segment sequence from the unit test) 8 gets added before 4 is sealed so that we make sure that when 4 is added after the seal of 1 is encountered that 4 does not replace 8 in the map.